File: hw3/hw31.js
use school;

db.students.find().forEach(function(student) {
	var s = Array();
	for ( var i = 0; i < student.scores.length; i++) {
		s.push(student.scores[i].score);
	}
	s_min = Math.min.apply(Math, s);
	db.students.update({
		_id : student._id
	}, {
		$pull : {
			scores : {
				score : s_min
			}
		}
	});
})

db.students.find({_id:100}).pretty();

db.students.aggregate({
	'$unwind' : '$scores'
}, {
	'$group' : {
		'_id' : '$_id',
		'average' : {
			$avg : '$scores.score'
		}
	}
}, {
	'$sort' : {
		'average' : -1
	}
}, {
	'$limit' : 1
})

File: hw3/hw31.py
import pymongo
import sys

connection = pymongo.MongoClient("mongodb://localhost")
db = connection.school
students = db.students


def average(students):
    for student in students.find() :
        homework_table = [ float(s['score']) for s in student['scores'] if s['type'] == 'homework']
        min_score = min(homework_table)
        reduced_score_table = [h for h in homework_table if h > min_score]
        #students.update({'_id':student['_id']},{'$pull':{'scores.score':min_score}})
        avg=(sum(reduced_score_table)/len(reduced_score_table))
        yield(student['_id'],avg) 

#print(sorted(average(students),key=lambda k:k[1],reverse=True))

def lowest_homework(students=students):
    for student in students.find() :
        homework_table = [ float(s['score']) for s in student['scores'] if s['type'] == 'homework']
        min_score = min(homework_table)
        yield(student['_id'],min_score)
        
         

for (s_id,min_score) in lowest_homework(students):
    print(str(s_id)+" "+str(min_score))
    students.update({'_id':s_id},{'$pull':{'scores.score':min_score}})




        
